nntrainer/layers/cl_layers/concat_cl.cpp

```diff
@@ -0,0 +1,540 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024  Niket Agarwal <niket.a@samsung.com>
+ *
+ * @file   concat_cl.cpp
+ * @date   2 July 2024
+ * @brief  Implementation of Concat Layer
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Niket Agarwal <niket.a@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#include <cstring>
+#include <vector>
+
+#include <concat_cl.h>
+#include <iostream>
+#include <layer_context.h>
+#include <nntr_threads.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <node_exporter.h>
+#include <tensor_dim.h>
+#include <util_func.h>
+
+std::string concat_cl_kernel_fp16_ =
```

Comment:
These kernels (fp15 and fp32) seem to be applicable for the case where its concat_dim = 3 (axis = width); it does not match with what you write in `finalize` method. Please update this to make the code clear to understand.

Reply:
Made some changes, added the other two axes support as well. Please have a look and let me know what you think.